builder/qemu/step_type_boot_command.go

```diff
@@ -105,35 +119,61 @@ func (s *stepTypeBootCommand) Run(ctx context.Context, state multistep.StateBag)
 
 	d := bootcommand.NewVNCDriver(c, config.VNCConfig.BootKeyInterval)
 
-	ui.Say("Typing the boot command over VNC...")
-	command, err := interpolate.Render(config.VNCConfig.FlatBootCommand(), &configCtx)
-	if err != nil {
-		err := fmt.Errorf("Error preparing boot command: %s", err)
-		state.Put("error", err)
-		ui.Error(err.Error())
-		return multistep.ActionHalt
-	}
+	ui.Say("Typing the boot commands over VNC...")
 
-	seq, err := bootcommand.GenerateExpressionSequence(command)
-	if err != nil {
-		err := fmt.Errorf("Error generating boot command: %s", err)
-		state.Put("error", err)
-		ui.Error(err.Error())
-		return multistep.ActionHalt
-	}
+	for _, step := range bootSteps {
+		if len(step) == 0 {
+			continue
+		}
 
-	if err := seq.Do(ctx, d); err != nil {
-		err := fmt.Errorf("Error running boot command: %s", err)
-		state.Put("error", err)
-		ui.Error(err.Error())
-		return multistep.ActionHalt
-	}
+		var description string
+
+		if len(step) >= 2 {
+			description = step[1]
+		} else {
+			description = ""
+		}
+
+		if len(description) > 0 {
+			log.Printf("Typing boot command for: %s", description)
```

Comment:
I wonder, could this be a `Ui.Say`? If we're going through the trouble of documenting each command typed, might as well be notified at which step we're on all the time, not only if we've enabled verbose logs. What do you think?

Reply:
Fixed in a62cb69e281ef0f1d0e6695e3b5e801cf8b544b8.